### PR TITLE
Expose product subscription check #361

### PR DESCRIPTION
**Pull request by sa-mao**
Supersedes #360 due to Artifactory credentials not being propagated for PR from repository fork.

### CODE REVIEW

File: core/src/main/java/com/schibsted/account/network/response/ProductSubscription.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.schibsted.account.network.response
+
+data class ProductSubscription(
+    val name: String,
+    val code: Int,
+    val data: Data
+) {
+    data class Data(
```
Comment (zamzterz) [inline core/src/main/java/com/schibsted/account/network/response/ProductSubscription.kt]: Rename
```suggestion
    data class ProductAccess(
```

File: core/src/main/java/com/schibsted/account/network/service/user/UserContract.kt [hunk 2]
```diff
@@ -46,8 +47,12 @@ internal interface UserContract {
     @GET("api/2/user/{userId}/subscriptions")
     fun subscriptions(@Header(KEY_AUTHORIZATION) userBearer: String, @Path(KEY_USER_ID) userId: String): Call<ListContainer<Subscription>>
 
+    @GET("api/2/user/{userId}/product/{productId}")
+    fun getProductSubscription(@Header(KEY_AUTHORIZATION) userBearer: String, @Path(KEY_USER_ID) userId: String, @Path(KEY_PRODUCT_ID) productId: String): Call<ProductSubscription>
```
Comment (zamzterz) [inline core/src/main/java/com/schibsted/account/network/service/user/UserContract.kt]: Rename:
```suggestion
    fun getProductAcess(@Header(KEY_AUTHORIZATION) userBearer: String, @Path(KEY_USER_ID) userId: String, @Path(KEY_PRODUCT_ID) productId: String): Call<ProductSubscription>
```

File: core/src/main/java/com/schibsted/account/session/Profile.kt [hunk 3]
```diff
@@ -66,4 +67,20 @@ class Profile(val user: User, private val userService: UserService = UserService
                     callback.onSuccess(it.value)
                 }))
     }
+
+    fun getProductSubscription(productId: String, callback: ResultCallback<ProductSubscription>) {
```
Comment (zamzterz) [inline core/src/main/java/com/schibsted/account/session/Profile.kt]: Rename:
```suggestion
    fun getProductAccess(productId: String, callback: ResultCallback<ProductSubscription>) {
```

File: core/src/main/java/com/schibsted/account/session/Profile.kt [hunk 4]
```diff
@@ -66,4 +68,20 @@ class Profile(val user: User, private val userService: UserService = UserService
                     callback.onSuccess(it.value)
                 }))
     }
+
+    fun getProductAccess(productId: String, callback: ResultCallback<ApiContainer<ProductSubscription.ProductAccess>>) {
```
Comment (Nexcius) [inline core/src/main/java/com/schibsted/account/session/Profile.kt]: Kotlin 1.3 came with a new class; `SuccessOrFailure` I'd recommend migrating to this whenever possible (separate PR ofc)

**Comment by schibsted-rafal-adamek** [comment]
@zamzterz, can you re-review? :) It looks like your requests have been adressed.

**Pull request merged**